src/pudl/transform/ferc1.py

```diff
@@ -853,6 +964,228 @@ def strip_col_of_non_ints(self, col: pd.Series) -> pd.DataFrame:
+            row_map.sched_table_name == self.source_table_id(Ferc1Source.DBF),
+            ["report_year", "row_number", "xbrl_column"],
+        ]
+
+        # Create an index containing all possible combinations of years and row numbers
+        idx = pd.MultiIndex.from_product(
+            [Ferc1Settings().dbf_years, row_map.row_number.unique()],
+            names=["report_year", "row_number"],
+        )
+
+        # Concatenate the row map with the empty index, so we have blank spaces to fill:
+        row_map = pd.concat(
+            [pd.DataFrame(index=idx), row_map.set_index(["report_year", "row_number"])],
+            axis="columns",
+        ).reset_index()
+
+        # Forward fill missing XBRL column names, until a new definition for the row
+        # number is found:
+        row_map["xbrl_column"] = row_map.groupby("row_number").xbrl_column.transform(
+            "ffill"
+        )
+        # Drop any rows that do not actually map between DBF rows and XBRL columns:
+        row_map = row_map.replace({"HEADER_ROW": np.nan}).dropna(subset=["xbrl_column"])
+        return row_map
+
+    @cache_df(key="dbf")
+    def process_dbf(self, raw_dbf: pd.DataFrame) -> pd.DataFrame:
+        """Process DBF inputs, aligning historical row numbers to XBRL account IDs."""
+        logger.info(f"{self.table_id.value}: Processing DBF data pre-concatenation.")
+        return (
+            self.drop_footnote_columns_dbf(raw_dbf)
+            # Note: in this rename_columns we have to pass in params, since we're using
+            # the inherited method, with param specific to the child class.
+            .pipe(self.rename_columns, params=self.params.rename_columns_ferc1.dbf)
+            .pipe(self.assign_record_id, source_ferc1=Ferc1Source.DBF)
+            .pipe(self.align_row_numbers_dbf)
+            .pipe(self.drop_unused_original_columns_dbf)
+            .pipe(
+                self.assign_utility_id_ferc1,
+                source_ferc1=Ferc1Source.DBF,
+            )
+        )
+
+    @cache_df(key="dbf")
+    def align_row_numbers_dbf(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Align historical FERC1 DBF row numbers with XBRL account IDs."""
+        return pd.merge(
+            df,
+            get_ferc1_dbf_xbrl_glue(self.source_table_id(Ferc1Source.DBF)),
+            on=["report_year", "row_number"],
+        ).rename(columns={"xbrl_column_stem": "ferc_account_id"})
+
+    @cache_df(key="xbrl")
+    def process_xbrl(
+        self,
+        raw_xbrl_instant: pd.DataFrame,
+        raw_xbrl_duration: pd.DataFrame,
+    ) -> pd.DataFrame:
+        """XBRL-specific transformations that take place before concatenation."""
+        logger.info(f"{self.table_id.value}: Processing XBRL data pre-concatenation.")
+        return (
+            self.merge_instant_and_duration_tables_xbrl(
+                raw_xbrl_instant, raw_xbrl_duration
+            )
+            .pipe(self.tidy_ferc_accounts_xbrl)
```

Comment:
Similar to `process_dbf()` above, the only line that differs from the inherited method here is the call to `tidy_ferc_accounts_xbrl()`. What's the right structure to avoid the duplication of everything else in the method? Again I guess we could define it to do nothing by default and have it get called in all cases. But it should probably be a more generic `tidy_wide_xbrl()` since different tables need different reshaping operations.

Reply:
I've updated the inherited method to make this call, but have it do nothing unless the child class has implemented it for the tables that need reshaping. See `wide_to_tidy_xbrl()`